go/vt/vtgate/evalengine/expr_logical.go

```diff
@@ -691,6 +692,8 @@ func (cs *CaseExpr) compile(c *compiler) (ctype, error) {
 		}
 
 		ta.add(then.Type, then.Flag)
+		scale = max(scale, then.Scale)
+		size = max(size, then.Size)
```

Comment:
Does this logic apply to other usages of the `typeAggregation` as well? And should this really be part of `typeAggregation` therefore instead? Seems like that might be more appropriate to refactor it into that?

Reply:
I have made the required changes. I wasn't sure what the length and scale for other data types looks like, so I just passed 0 from everywhere.